cirrus/server/tests/test_main.py

```diff
@@ -523,7 +714,165 @@ def test_get_features_with_and_without_nimbus_preview(
+            "events": [
+                {
+                    "branch_slug": "treatment",
+                    "change": "Enrollment",
+                    "enrollment_id": "enrollment_id_2",
+                    "experiment_slug": "experiment_slug_2",
+                    "reason": None,
+                }
+            ],
+        }
+
+        # Mock collate_enrollment_metric_data to process events correctly
+        mock_collate_enrollment_metric_data.side_effect = (
+            lambda enrolled_partial_configuration, client_id, nimbus_preview_flag: [
+                EnrollmentMetricData(
+                    nimbus_user_id=client_id,
+                    app_id="test_app_id",
+                    experiment_slug=event["experiment_slug"],
+                    branch_slug=event["branch_slug"],
+                    experiment_type="experiment" if nimbus_preview_flag else "rollout",
+                    is_preview=nimbus_preview_flag,
+                )
+                for event in enrolled_partial_configuration["events"]
+            ]
+        )
+
+        # Mock feature configurations to return a simplified structure
+        mock_compute_feature_configurations.side_effect = (
+            lambda enrolled_partial_configuration: {
+                feature_id: feature_data["feature"]["value"]
+                for feature_id, feature_data in enrolled_partial_configuration[
+                    "enrolledFeatureConfigMap"
+                ].items()
+            }
+        )
+
+        # Test for live SDK (no nimbus_preview)
+        response = client.post("/v2/features/", json=request_data)
+        assert response.status_code == 200
+        assert response.json() == {
+            "Features": {
+                "example-feature": {"enabled": False, "something": "wicked"},
+            },
+            "Enrollments": [
+                {
+                    "nimbus_user_id": "4a1d71ab-29a2-4c5f-9e1d-9d9df2e6e449",
+                    "app_id": "test_app_id",
+                    "experiment": "experiment_slug_1",
+                    "branch": "treatment",
+                    "experiment_type": "rollout",
+                    "is_preview": False,
+                }
+            ],
+        }
+
+        # Test for preview (with nimbus_preview=true)
+        response = client.post("/v2/features/?nimbus_preview=true", json=request_data)
+        assert response.status_code == 200
+        assert response.json() == {
+            "Features": {
+                "example-feature": {"enabled": True, "something": "preview"},
+            },
+            "Enrollments": [
+                {
+                    "nimbus_user_id": "4a1d71ab-29a2-4c5f-9e1d-9d9df2e6e449",
+                    "app_id": "test_app_id",
+                    "experiment": "experiment_slug_2",
+                    "branch": "treatment",
+                    "experiment_type": "experiment",
+                    "is_preview": True,
+                }
+            ],
+        }
```

Comment:
In general we should try to keep tests to testing a single case in each one test.  This is testing two cases, we should split that into two separate tests.